stor/dx.py

```diff
@@ -144,41 +138,273 @@ def download_objects(self):  # may not need it
         raise NotImplementedError
 
     def remove(self):
-        raise NotImplementedError
+        """Removes a single object from DX platform
+
+        Raises:
+            ValueError: The path is invalid.
```

Comment:
If the file doesn't exist, a NotFoundError is raised when trying to access its canonical_resource property. 
```file_handler = dxpy.DXFile(dxid=self.canonical_resource,
                                   project=self.canonical_project)```
dxpy doesn't get a chance to raise an exception. However, as stated in your summary, I should wrap this with a context manager for other scenarios.

Reply:
When you attempt in dx toolkit, it will error: `ResolutionError: Unable to resolve "file" to a data object or folder name in '/parent'`. When you attempt in stor, it will error: `NotFoundError: No data object was found for the given path on DNAnexus`

Reply:
Yes, still the same behavior in dxpy/dxpy3 and stor